pennylane/ops/qutrit/matrix_ops.py

```diff
@@ -98,20 +109,214 @@ def compute_matrix(U):  # pylint: disable=arguments-differ
+               [ 0.        +0.j,  0.        +0.j,  0.        +0.j,  0.        +0.j,  0.        +0.j,  0.        +0.j,  0.        +0.j,  1.        +0.j,  0.        +0.j],
+               [ 0.        +0.j,  0.        +0.j,  0.        +0.j,  0.        +0.j,  0.        +0.j,  0.        +0.j,  0.        +0.j,  0.        +0.j,  1.        +0.j]])
+        """
+        target_dim = 3 ** len(u_wires)
+        shape = qml.math.shape(U)
+        if not (len(shape) in {2, 3} and shape[-2:] == (target_dim, target_dim)):
+            raise ValueError(
+                f"Input unitary must be of shape {(target_dim, target_dim)} or "
+                f"(batch_size, {target_dim}, {target_dim})."
+            )
+
+        # A multi-controlled operation is a block-diagonal matrix partitioned into
+        # blocks where the operation being applied sits in the block positioned at
+        # the integer value of the control string.
+
+        total_wires = qml.wires.Wires(control_wires) + qml.wires.Wires(u_wires)
+
+        # if control values unspecified, we control on the all-twos string
+        if not control_values:
+            control_values = "2" * len(control_wires)
+
+        if isinstance(control_values, str):
+            if len(control_values) != len(control_wires):
+                raise ValueError(
+                    "Length of control trit string must equal number of control wires."
+                )
+
+            # Make sure all values are either 0 or 1 or 2
+            if any(x not in ["0", "1", "2"] for x in control_values):
+                raise ValueError("String of control values can contain only '0' or '1' or '2'.")
+
+            control_int = int(control_values, 3)
+        else:
+            raise ValueError("Alternative control values must be passed as a ternary string.")
+
+        padding_left = control_int * target_dim
+        padding_right = 3 ** len(total_wires) - target_dim - padding_left
+
+        interface = qml.math.get_interface(U)
+        left_pad = qml.math.cast_like(qml.math.eye(padding_left, like=interface), 1j)
+        right_pad = qml.math.cast_like(qml.math.eye(padding_right, like=interface), 1j)
+        if len(qml.math.shape(U)) == 3:
+            return qml.math.stack([qml.math.block_diag([left_pad, _U, right_pad]) for _U in U])
+        return qml.math.block_diag([left_pad, U, right_pad])
+
+    @property
+    def control_wires(self):
+        return self.hyperparameters["control_wires"]
+
+    def pow(self, z):
+        if isinstance(z, int):
+            return [
+                ControlledQutritUnitary(
+                    qml.math.linalg.matrix_power(self.data[0], z),
+                    control_wires=self.control_wires,
+                    wires=self.hyperparameters["u_wires"],
+                )
+            ]
+        return super().pow(z)
+
+    def _controlled(self, wire):
```

Comment:
I notice this looks a bit different than the same method in `ControlledQubitUnitary`, we're not considering any inverses here. Is there a reason this is different?

Reply:
`_controlled` for `ControlledQubitUnitary` used to be different. I think it got updated recently because all the design decisions I made for `ControlledQutritUnitary` were based on `ControlledQubitUnitary`.